test/dummy.jl

```diff
@@ -1,9 +1,19 @@
-struct DummyModel <: MOI.ModelLike
+# DummyModel{true} implements add_variable and add_constraint
+struct DummyModel{A} <: MOI.ModelLike
```

Comment:
What about an abstract dummy model and two concrete subclasses to distinguish between the versions that implement `add_variable` or not.